src/modules/reporting/helpers/progress-calculations.js

```diff
@@ -2,45 +2,62 @@ import { formatAttoRep } from 'utils/format-number'
+
+const ONE = new BigNumber(1)
+const ONE_HUNDRED = new BigNumber(100)
+
+// Percentages are ALWAYS in number range so results are returned as number
+export const calculatePercentage = (numSize, numTotalStake) => {
+  if (numSize === null || numTotalStake === null) return 0
+  if (typeof numSize === 'undefined' || typeof numTotalStake === 'undefined') return 0
+
+  // NB: Remove whence paramaters are converted outside
+  const size = new BigNumber(numSize, 10)
+  const totalStake = new BigNumber(numTotalStake, 10)
+  // NB: Remove whence paramaters are converted outside
+
+  if (size.isEqualTo(0)) return 0
+  if (size.lt(0) || totalStake.lt(0)) return 0
+  if (totalStake.isEqualTo(0)) return 0
+
+  const ratio = size.minus(totalStake).dividedBy(size)
+  return (ONE.minus(ratio).times(ONE_HUNDRED)).integerValue().toNumber()
 }
 
-export const calculateNonAccountPercentage = (size, stakeCurrent, accountStakeCurrent) => {
-  const result = calculatePercentage(size, new BigNumber(stakeCurrent).minus(new BigNumber(accountStakeCurrent)))
-  return result
+export const calculateNonAccountPercentage = (size, numStakeCurrent, numAccountStakeCurrent) => {
+  // NB: Remove whence paramaters are converted outside
+  const stakeCurrent = new BigNumber(numStakeCurrent, 10)
+  const accountStakeCurrent = new BigNumber(numAccountStakeCurrent, 10)
+  // NB: Remove whence paramaters are converted outside
+
+  return calculatePercentage(size, stakeCurrent.minus(accountStakeCurrent))
 }
 
-export const calculateAddedStakePercentage = (size, accountStake, addedStake) => {
-  const attoRep = convertRepToAttoRep(addedStake)
-  const addedAccountStake = new BigNumber(accountStake).plus(new BigNumber(attoRep))
-  const result = calculatePercentage(size, new BigNumber(addedAccountStake).toNumber())
-  return result
+export const calculateAddedStakePercentage = (size, numAccountStake, numAddedStake) => {
+  // NB: Remove whence paramaters are converted outside
+  const accountStake = new BigNumber(numAccountStake, 10)
+  const addedStake = new BigNumber(numAddedStake, 10)
+  // NB: Remove whence paramaters are converted outside
+
+  return calculatePercentage(size, accountStake.plus(convertRepToAttoRep(addedStake)))
 }
 
-export const calculateTentativeRemainingRep = (size, totalStake, tentativeStake) => {
-  const attoRep = convertRepToAttoRep(tentativeStake)
-  const result = calculateRemainingValue(size, new BigNumber(totalStake).plus(new BigNumber(attoRep)).toNumber())
-  if (result <= 0) return '0'
+export const calculateTentativeRemainingRep = (size, numTotalStake, numTentativeStake) => {
+  // NB: Remove whence paramaters are converted outside
+  const totalStake = new BigNumber(numTotalStake, 10)
+  const tentativeStake = new BigNumber(numTentativeStake, 10)
+  // NB: Remove whence paramaters are converted outside
+
+  const result = calculateRemainingValue(size, totalStake.plus(convertRepToAttoRep(tentativeStake))).toNumber()
```

Comment:
found small issue where `calculateRemainingValue` expects size to be BigNumber. I'll fix.